### PR TITLE
 🛠  DB config refactor - env vars & deployment config & 🔨 Security vuln fixes

### PR DESCRIPTION
**Pull request by hirako2000**
### Motivation and Context

- Won't only run this locally, env variables are convenient when deploying to fly.io
- Code has vulnerabilities and other coding best practice issues caught by static analysis tools

### Description 

- Replaces db name, those references of feathers
- Makes use of env variables for DB URL
- Make model IDs types and default value consistent
- Fixes vulnerability in minimist and uglifyjs
- Fixes the privilege escalation potential and writing to root fs in Docker containers
- Adds clean script, fixes the CI job
- Adds fly.toml. and cleans up Docker
- -Adds dockerignore
-  Updates README and remove redundant steps

### CODE REVIEW

**Comment by hirako2000** [comment]
this PR is enormous and is a good example of PR to not create. will keep things smaller next time.

File: src/sequelize.ts [hunk 1]
```diff
@@ -2,29 +2,39 @@ import { Sequelize } from "sequelize";
 import type { Application } from "./declarations";
 import logger from "./logger";
 
-export function setupSequelize (app: Application): void {
+export function setupSequelize(app: Application): void {
+	const dialect = "postgres"; // Or your dialect name
```
Comment (bachiro276) [inline src/sequelize.ts:6]: Can we rename this variable to something more descriptive?

**Review by bachiro276 — APPROVED** [review]
Everything is very clear and intuitive. I understood almost all the code base!

**Comment by hirako2000** [comment]
Awesome!

Thanks for going through it. 

Let's let the author of PRs to merge them :)

File: src/sequelize.ts (hunk 1 above)
Comment (hirako2000) [reply src/sequelize.ts:6]: Hmm I took that from the official doc. I think.

Not sure what to name it. `queryDialect`?

**Pull request merged**